### ISSUE TITLE
convert float to string within numba function

### ISSUE THREAD

**Issue by elChapoSing** [numba/numba#4758]
## Reporting a bug

- [x] I am using the latest released version of Numba
![image](https://user-images.githubusercontent.com/5416611/67744862-a9d93e00-fa5d-11e9-9d3a-05efee1f89dc.png)
- [x] I have included below a minimal working reproducer.

```pyton
import numpy as np
import numba as nb

@nb.njit(nogil=True)
def func():
    my_array = np.empty(6, dtype=np.dtype("U20"))
    my_array[0] = np.str(2.35646)
    return my_array


if __name__ == '__main__':
    a = func()
    print(a)
```

I would have expected the conversion to go without trouble but I'm getting this error message
```
numba.errors.TypingError: Failed in nopython mode pipeline (step: nopython frontend)
Invalid use of Function(<class 'str'>) with argument(s) of type(s): (float64)
```

**Comment by stuartarchibald** [comment]
Thanks for the report. In general only a subset of NumPy and Python functions/features are supported by Numba, documentation on those supported can be found here.
 * [NumPy Features](http://numba.pydata.org/numba-doc/latest/reference/numpysupported.html)
 * [Python Features](http://numba.pydata.org/numba-doc/latest/reference/pysupported.html)

Specifically, `np.str()` and `str(int)` are not yet supported.

**Comment by elChapoSing** [comment]
Ok noted. Thank you for taking the time to reply.

How would one achieve what I need if at all possible?

I'm trying to calculate some numbers and output a mixed-type array with
strings and floats. I wanted to convert everything to string because mixed
type are not supported in numpy.

**Comment by stuartarchibald** [comment]
> Ok noted. Thank you for taking the time to reply. How would one achieve what I need if at all possible?

It's simply not possible to do this at present in Numba. It's possible to add support for this to Numba but it will likely be quite challenging.

> I'm trying to calculate some numbers and output a mixed-type array with strings and floats. I wanted to convert everything to string because mixed type are not supported in numpy.

I think you may end up with some code that doesn't have great performance if you opt to use strings as your representation. Does the array really need to be mixed? Could it be two separate data sets? Is an array the best container to store this information?

**Comment by elChapoSing** [comment]
You are right of course.

It's dates in fact so I could convert them to float : does numba allow
that?

**Comment by stuartarchibald** [comment]
What sort of dates? `np.datetime64`s?

**Comment by elChapoSing** [comment]
I can still decide which type of dates at this point. Anything works
really.

**Comment by LeslieGerman** [comment]
Well in fact none of the possible python methods for float -> string conversion work:
- str(3.14)
- repr(3.14)
- format(3.14, 'f')
- numpy.format_float_positional(3.14)
:|
So I support this feature request.

**Comment by shaunc** [comment]
Yes ... for me would be great to have some support even if not fast. (Ok ... could write my own...) I'm looking to write a number into an error message inside a jitted function that may be inside bunch of other jitted functions. I can rewrite return values to bubble the error up but it will make a mess and make code more complex and slower on the average for the rare case of an error.
